pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/GrowableArrayBlockingQueue.java

```diff
@@ -436,6 +443,14 @@ public void terminate(@Nullable Consumer<T> itemAfterTerminatedHandler) {
         } finally {
             tailLock.unlockWrite(stamp);
         }
+
+        // Signal waiting consumer threads to prevent indefinite blocking after termination
+        headLock.lock();
+        try {
+            isNotEmpty.signal();
```

Comment:
We should wake up all waiting threads.
```suggestion
            isNotEmpty.signalAll();
```
Please add a test case for more than one thread so that this case is also covered.

Reply:
I guess it would be extraordinary to use the Consumer API this way, but that's possible and the API doesn't prevent multiple threads to call `.receive()` at the same time.